src/components/account/Quota.vue

```diff
@@ -0,0 +1,109 @@
+<!--
+  - SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-FileCopyrightText: 2024 STRATO AG
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<section class="section">
+		<h2>{{ t('simplesettings', 'Storage usage') }}</h2>
+		<div class="quota">
+			<NcProgressBar
+				v-if="hasLimitedSpace"
+				size="medium"
+				:value="usageRelative"
+				:color="barColor"
+				:style="{'background-color': backgroundColor}" />
+			<div class="quota-info">
+				<p>
+					<span v-if="hasLimitedSpace" :style="quotaPillStyles(barColor)" />
+					<span v-html="quotaUsedText" />
+				</p>
+				<p v-if="hasLimitedSpace">
+					<span :style="quotaPillStyles(backgroundColor)" />
+					<span v-html="quotaFreeText" />
+				</p>
+			</div>
+		</div>
+	</section>
+</template>
+
+<script>
+import { loadState } from '@nextcloud/initial-state'
+import NcProgressBar from '@nextcloud/vue/dist/Components/NcProgressBar.js'
+
+/** SYNC to be kept in sync with `lib/public/Files/FileInfo.php` */
+const SPACE_UNLIMITED = -3
+
+const barColor = null
+const backgroundColor = null
+
+const { quota, freeSpace, usage, usageRelative } = loadState('simplesettings', 'personalInfoParameters', {})
+const hasLimitedSpace = quota !== SPACE_UNLIMITED
+
+export default {
+	name: 'Quota',
+
+	components: {
+		NcProgressBar,
+	},
+
+	data() {
+		return {
+			usageRelative,
+			barColor,
+			backgroundColor,
+			hasLimitedSpace,
+		}
+	},
+
+	mounted() {
+		const styles = getComputedStyle(document.documentElement)
+		this.barColor = styles.getPropertyValue('--ion-color-blue-b4')
+		this.backgroundColor = styles.getPropertyValue('--ion-color-cool-grey-c2')
+	},
+
+	computed: {
+		quotaUsedText() {
+			return t(
+				'settings', 'Used: <strong>{usage}</strong>', { usage },
+			)
+		},
+		quotaFreeText() {
+			return t(
+				'settings', 'Free: <strong>{freeSpace}</strong>', { freeSpace },
+			)
+		},
+	},
+
+	methods: {
+		quotaPillStyles(color) {
+			return {
+				'background-color': color,
+				'border-radius': 10 + 'px',
+				'margin-right': 12 + 'px',
+				display: 'inline-block',
+				height: 10 + 'px',
+				width: 20 + 'px',
+			}
```

Comment:
question: is there a reason not to write the numbers directly into the string
```suggestion
			return {
				'background-color': color,
				'border-radius': '10px',
				'margin-right': '12px',
				display: 'inline-block',
				height: '10px',
				width: '20px',
			}
```

Reply:
No... just personal preference. I don't think it makes any difference. :shrug:

Reply:
✅